lib/agent.js

```diff
@@ -257,3 +257,52 @@ Agent.prototype.handleUncaughtExceptions = function (cb) {
 Agent.prototype.flush = function (cb) {
   this._instrumentation.flush(cb)
 }
+
+Agent.prototype.lambda = function wrapLambda (name, fn) {
+  var agent = this
+  return function wrappedLambda (payload, context, callback) {
+    var trans = agent.startTransaction()
+    if (!trans) {
+      return fn(payload, context, callback)
+    }
+
+    trans.type = 'lambda'
+    trans.name = name
+
+    function wrapLambdaCallback (callback) {
+      return function wrappedLambdaCallback (err, result) {
+        var context = this
+
+        trans.setCustomContext({
+          input: payload,
+          output: result
+        })
+
+        if (err) {
+          agent.captureError(err, done)
+        } else {
+          setImmediate(done)
```

Comment:
What's your reason for using `setImmediate` instead of `process.nextTick`? Just curious 😃

Reply:
I just generally avoid using `process.nextTick` because they run before  I/O callbacks, so they can potentially block I/O callbacks indefinitely if triggered recursively. I doubt that's a concern here though.

Reply:
Got'ya 👍